subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java

```diff
@@ -369,6 +370,38 @@ public String determineName(Configuration c) {
     Configuration defaultDependencies(Action<? super DependencySet> action);
 
     /**
+     * Execute the given action before the configuration first participates in
+     * dependency resolution. A {@code Configuration} will participate in dependency resolution
+     * when:
+     * <ul>
+     *     <li>The {@link Configuration} itself is resolved</li>
+     *     <li>Another {@link Configuration} that extends this one is resolved</li>
+     *     <li>Another {@link Configuration} that references this one as a project dependency is resolved</li>
+     *     <li>The {@link Configuration} is referenced in publishing</li>
```

Comment:
❌ If this is already true, there should be a test showing it.

Reply:
You're right, I added this prematurely (even though it probably does work). 
I'll remove this line and re-add as part of #3247.